### DISCUSSION TITLE
how to debug  gltf-transform optimize

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [donmccurdy/glTF-Transform#1520] [Q&A]
the glb model without animation loads just fine, however when I apply the external animation to it, I seen only part of the model. The animation glb file is fine, as it does not seem to have issues applied to a different model.

ps I assume I can put all these png files in the same folder, they don't cause any weird transformations if I accidentally overwrite a file from a different model?

**Comment by donmccurdy** [comment]
Hi @f1-outsourcing!

The 'optimize' command is an aggressive set of optimizations, including flattening the scene hierarchy and joining/merging meshes together. For example, it may reduce the scene from 100 meshes to 1 mesh. If you're planning to apply an animation later that would move meshes individually, then you'll want to adjust the options (`gltf-transform optimize --help`) to preserve the meshes and the node structure, or use the other more focused CLI commands individually, or embed the animation in the file being optimized.

If two glTF models referencing the PNG textures with the same name are written to the same folder, the PNG will be overwritten by the second model. If the textures are identical, that's fine, and it can be helpful to share textures between models. If they're not identical, you need to either write them to separate folders, or else write to `.glb` which will embed the textures inside the model.
